### PR TITLE
Removed hard coded screen width and height

### PR DESCRIPTION
**Pull request by tsadarsh**
PR for Issue #6.

### CODE REVIEW

**Comment by nmccrea** [comment]
Awesome @maddypie thanks so much. Will take a look first thing tomorrow. This is def a problem.

**Review by nmccrea — CHANGES_REQUESTED** [review]
Alrighty @maddypie. Thanks so much for tackling this.

In general, we always want to keep the direct dependencies of any long-term project as few as possible, and to try to fix things using the tools we already have, since this minimizes complexity and potential points of failure, and makes the project easier to maintain in the long run. We should only add more dependencies if we can't do what we are trying to do with the existing dependencies.

By using Tkinter for this fix, we are making the software dependent on an additional GUI library. In this case, since we are already using a GUI library (Gtk), we want to see if there is a solution to this problem using Gtk. It took me a minute to figure out but it looks like the most up-to-date way to do this is to use [Gdk 3](https://lazka.github.io/pgi-docs/Gdk-3.0/index.html), a low-level library that comes with Gtk 3. Here are the changes I propose for `viewer.py`:

```python
import gi
gi.require_version('Gtk', '3.0')
from gi.repository import Gtk as gtk

gi.require_version('Gdk', '3.0')
from gi.repository import Gdk

from gui.frame import Frame
from gui.painter import Painter

display = Gdk.Display.get_default()
if not display:
  exit("GUI code not find the display.")

MONITOR_GEOMETRY = display.get_monitor(0).get_geometry()
MONITOR_WIDTH = MONITOR_GEOMETRY.width
MONITOR_HEIGHT = MONITOR_GEOMETRY.height

DEFAULT_VIEW_PIX_W = round(MONITOR_WIDTH * 0.5)   # pixels
DEFAULT_VIEW_PIX_H = round(MONITOR_HEIGHT * 0.7)  # pixels
DEFAULT_ZOOM = 100                                # pixels per meter

# ...
```

It's slightly more verbose but it keeps the solution within the existing ecosystem, which is a priority for me.

The Tkinter solution also does not run on my Mac as it is written. I have not gotten to the bottom of it but it is a platform-specific issue and a big ol' rabbit hole I'd rather not go down unless I have to.

Since it looks like you are using Linux, would you be willing to try out the above solution and update the PR if it works on your platform? Alternatively, you can [give me edit access to this PR](https://github.blog/2016-09-07-improving-collaboration-with-forks/) and I can push it myself, but I would love to have your confirmation that it works on both systems.

### Longer Term

I am still getting back up to speed when it comes to all things Python and desktop dev, but a quick google of popular Python GUI libraries reveals that Gtk does not even make the list for most folks these days (while Tk does). For this and several other reasons, I think a major refactor of this project's UI may be in its future, during which we may ditch Gtk altogether. More on this to come. Such a transition should be planned and executed carefully though, so in the meantime, I want to cling to Gtk as closely as possible until we are ready to make that leap.

**Comment by tsadarsh** [comment]
@nmccrea, yes, sticking to ```gtk``` library is the best choice for now.  I have made the suggested changes to ```viewer.py```. I think I have already given you access to edit my PR. 
I now know that ```tkinter``` is highly platform specific. Thanks for that pro-tip.
Yes I am using Linux and it works properly in my OS :+1: .

File: gui/viewer.py [hunk 1]
```diff
@@ -24,12 +24,24 @@
 gi.require_version('Gtk', '3.0')
 from gi.repository import Gtk as gtk
 
+gi.require_version('Gdk', '3.0')
+from gi.repository import Gdk as gdk
```
Comment (nmccrea) [inline gui/viewer.py]: If you backmerge in the latest from `master` you will see I got rid of the renaming of `Gtk` to `gtk` (line 25 above). I don't see a good reason for this rename and it's basically an artifact from porting everything to Python 3. For consistency let's stick with `Gdk`, and then I think this PR will be good to go.

**Review by nmccrea — CHANGES_REQUESTED** [review]

File: gui/viewer.py (hunk 1 above)
Comment (tsadarsh) [reply gui/viewer.py]: I am sorry. My action was motivated by **line 25**. 
Back-merging **done** :+1: .

**Review by nmccrea — APPROVED** [review]

**Comment by nmccrea** [comment]
Awesome work @maddypie thank you!

**Pull request merged**